plugins/main/public/components/common/data-grid/data-grid-service.ts

```diff
@@ -174,24 +174,29 @@ export const parseColumns = (
   defaultColumns: tDataGridColumn[] = [],
 ): EuiDataGridColumn[] => {
   // remove _source field becuase is a object field and is not supported
-  fields = fields.filter(field => field.name !== '_source');
   // merge the properties of the field with the default columns
+  if (!fields?.length) {
+    return defaultColumns;
```

Comment:
**thought**: if I am not wrong, the `fields` and `defaultColumns` parameters has different type and regarding the business logic of the function this could return different types. I am not sure if this is intentional or could cause other problems.

Reply:
I've changed the typo. The parseColumns method must return a tDataGridColumn array. In that typo only the id property is required.